include/criterion/assert.h

```diff
@@ -157,19 +1094,144 @@
 
 # ifdef __cplusplus
 
+/**@}*/
+
+/**
+ * @defgroup ExceptionAsserts Exception asserts
```

Comment:
Same as the safe array assertions, there should be a notice that these are C++-only, even thought it might be obvious.